types/errors.go

```diff
@@ -73,40 +78,43 @@ func CodeToDefaultMsg(code CodeType) string {
 
 // nolint
 func ErrInternal(msg string) Error {
-	return newError(CodeInternal, msg)
+	return newError(CodeInternal, msg, []cmn.KVPair{})
```

Comment:
fine, but in future feel free to avoid this by introducing `newErrorWithTags` function instead ...

Reply:
typing []cmn.KVPair{} is unnecessary, we should just pass in nil.

Reply:
They're not identical (== and reflect.DeepEqual etc will say that they are different), but len(x) behaves identically so looping over has the same behavior.

Reply:
Even if we needed an empty list a opposed to nil, the place to type []cmn.KVPair{} is in newError().

Reply:
Also, while Tendermint can support cmn.KVPair{}, for the SDK I suggest we use a []SSPair{}, where SSPair is struct{string,string}.  At some point we need to display these tags, and I imagine life will be easier for the Voyager team if all keys and values were guaranteed to be unicode strings.  Of course a `string` isn't guaranteed to be valid unicode, but we can enforce that somewhere, e.g. Voyager can do the conversion and display '?' otherwise.

Otherwise we're forced to display hex bytes.  Not a good user experience...

Reply:
Tendermint ABCI can stay cmn.KVPair, as we don't know how other SDKs will use Tendermint, and besides, KVPair still works as ABCI transport.